### PR TITLE
Rewrite the Pipelines documentation for clarity and flow.

### PR DESCRIPTION
**Pull request by sergetron**
Rewrites the Pipelines documentation for clarity and flow.

### CODE REVIEW

File: docs/pipelines.md [hunk 1]
```diff
@@ -319,21 +285,17 @@ For example see this `Pipeline` spec:
...
-is no output from `test-app`, so `build-app` uses `runAfter` to indicate that
-`test-app` should run before it, regardless of the order they appear in the
-spec.
+### Using the `retries` parameter
 
-#### retries
+For each `Task` in the `Pipeline` you can specify the number of times its
+execution should be retried if it fails. Any issues the `Task` is expected
+to encounter, such as network connectivity or missing dependencies, must
+return `False` for the "TaskRun Status Succeeded" `Condition`. If you don't
```
Comment (sbwsg) [inline docs/pipelines.md:293]: Suggest rewording the sentence about TaskRun Status Succeeded a little bit:

> When a Pipeline Task ends in failure, its `TaskRun` will have a `ConditionSucceeded` with value `False`.
The `retries` parameter allows the `Task` to be reattempted when this happens. This can be useful to set when you know there are likely to be issues with network connectivity or missing dependencies during Task execution.

Comment (sbwsg) [reply docs/pipelines.md:293]: Sorry, should be "`Succeeded` Condition" instead of "`ConditionSucceeded`".

File: docs/pipelines.md [hunk 2]
```diff
@@ -6,85 +6,68 @@ weight: 3
...
-        it fails. Could be a network error or a missing dependency. It does not
-        apply to cancellations.
-      - [`conditions`](#conditions) - Used when a task is to be executed only if the specified
-        conditions are evaluated to be true.
-      - [`timeout`](#timeout) - Specifies timeout after which the `TaskRun` for a Pipeline Task will
-        fail. There is no default timeout for a Pipeline Task timeout. If no timeout is specified for
-        the Pipeline Task, the only timeout taken into account for running a `Pipeline` will be a
-        [timeout for the `PipelineRun`](https://github.com/tektoncd/pipeline/blob/master/docs/pipelineruns.md#syntax).
-  - [`results`](#pipeline-results) - Specifies which `results` is defined for the pipeline
+      - [`from`](#using-the-from-parameter) - Indicates the data from a [`PipelineResource`](resources.md)
```
[the remark this answers is not in the document]
Comment (sergetron) [reply docs/pipelines.md:51]: Done and done.

File: docs/pipelines.md [hunk 3]
```diff
@@ -6,85 +6,68 @@ weight: 3
...
+        originates from the output of a previous `Task`.
+      - [`runAfter`](#using-the-runafter-parameter) - Indicates that a `Task`
+        should execute after another `Task` without output linking.
```
[the remark this answers is not in the document]
Comment (sergetron) [reply docs/pipelines.md:54]: Reworded to "one or more other `Tasks`".

File: docs/pipelines.md (hunk 1 above)
Comment (sergetron) [reply docs/pipelines.md:293]: Reworded.

File: docs/pipelines.md [hunk 4]
```diff
@@ -319,21 +285,21 @@ For example see this `Pipeline` spec:
...
-Sometimes you need a policy for retrying tasks which have problems such as
-network errors, missing dependencies or upload problems. Any of those issues must
-be reflected as False (corev1.ConditionFalse) within the TaskRun Status
-Succeeded Condition. For that reason there is an optional attribute called
-`retries` which declares how many times that task should be retried in case of
-failure.
+If you know that there will be issues with network connectivitity or missing
+dependencies when a `Task` executes, set its `retries` parameter to a suitable
+value greater than 0. If you don't explicitly specify a value, Tekton does not
+attempt to execute the failed `Task` again.
```
Comment (sbwsg) [inline docs/pipelines.md:298]: nice.

File: docs/pipelines.md [hunk 5]
```diff
@@ -319,21 +285,21 @@ For example see this `Pipeline` spec:
...
+`TaskRun` sets its `Succeeded` `Condition` to `False`. The `retries` parameter
+instructs Tekton to retry executing the `Task` when this happens.
 
-Sometimes you need a policy for retrying tasks which have problems such as
-network errors, missing dependencies or upload problems. Any of those issues must
-be reflected as False (corev1.ConditionFalse) within the TaskRun Status
-Succeeded Condition. For that reason there is an optional attribute called
-`retries` which declares how many times that task should be retried in case of
-failure.
+If you know that there will be issues with network connectivitity or missing
```
Comment (sbwsg) [inline docs/pipelines.md:295]: Nit: suggest `If you know, for example, that there will be`

I think it's worth making clear that these aren't the only two conditions under which a user might want to retry a Task.  Just the two examples we have on hand :)

File: docs/pipelines.md (hunk 4 above)
Comment (sergetron) [reply docs/pipelines.md:298]: *bows*

File: docs/pipelines.md (hunk 5 above)
Comment (sergetron) [reply docs/pipelines.md:295]: Good call - made it clear it's an example.

**Comment by sbwsg** [comment]
Needs a rebase on top of master then 🤞 should be good to go!
